server/base/src/main/java/org/apache/accumulo/server/security/handler/ZKPermHandler.java

```diff
@@ -468,22 +456,19 @@ private void createTablePerm(String user, TableId table, Set<TablePermission> pe
    */
   private void createNamespacePerm(String user, NamespaceId namespace,
       Set<NamespacePermission> perms) throws KeeperException, InterruptedException {
-    synchronized (zooCache) {
```

Comment:
Do you happen to know why all these previous callers were synchronizing here and why it's safe to remove it? It seems like it might have something to do with the fact that this code is updating a path that should be cached, so maybe we want to make sure it's cleared after the update is done?

Reply:
In some of the cases the code was calling ZooCache.clear(path), and in others it was calling ZooCache.clear(). In all of the cases it seems to call clear, then add something. It looks like all of the security classes that are modified in this PR used the same pattern. I'm wondering if I might have to add a Lock in each of the classes on which to synchronize on to preserve the behavior.